sdk/ml/azure-ai-ml/azure/ai/ml/entities/_registry/registry_support_classes.py

```diff
@@ -25,13 +33,50 @@ def __init__(
         :param acr_account_sku: The storage account service tier. Currently
             only Premium is a valid option for registries.
         :type acr_account_sku: str
-        :param arm_resource_id: Resource ID of the ACR account.
-        :type arm_resource_id: str
+        :param arm_resource_id: Resource ID of the ACR account. 
+        :type arm_resource_id: str. Default value is None.
         """
         self.acr_account_sku = acr_account_sku
         self.arm_resource_id = arm_resource_id
 
+    # acr should technically be a union between str and SystemCreatedAcrAccount,
+    # but python doesn't accept self class references apparently.
+    # Class method instead of normal function to accept possible
+    # string input.
+    @classmethod
+    def _to_rest_object(cls, acr) -> RestAcrDetails:
+        if hasattr(acr, "acr_account_sku") and acr.acr_account_sku is not None:
+            # SKU enum requires input to be a capitalized word.
+            # Format input to be acceptable as long as spelling is correct.
+            acr_account_sku = acr.acr_account_sku.capitalize()
+            return RestAcrDetails(
+                system_created_acr_account=RestSystemCreatedAcrAccount(
+                    acr_account_sku=acr_account_sku,
+                )
+            )
+        else:
+            return RestAcrDetails(
+                user_created_acr_account=RestUserCreatedAcrAccount(
+                    arm_resource_id=RestArmResourceId(resource_id=acr))
+            )
 
+    @classmethod
+    def _from_rest_object(cls, rest_obj: RestAcrDetails) -> "Union[str, SystemCreatedAcrAccount]":
+        if not rest_obj:
+            return None
+        # TODO should we even bother check if both values are set and throw an error? This shouldn't be possible.
+        if hasattr(rest_obj, "system_created_acr_account") and rest_obj.system_created_acr_account is not None:
+            return SystemCreatedAcrAccount(
+                acr_account_sku=rest_obj.system_created_acr_account.acr_account_sku,
+                arm_resource_id=rest_obj.system_created_acr_account.arm_resource_id,
+            )
+        elif hasattr(rest_obj, "user_created_acr_account") and rest_obj.user_created_acr_account is not None:
+            return rest_obj.user_created_acr_account
+        else:
+            return None  # TODO should this throw an error instead?
+
+
+@experimental
 class SystemCreatedStorageAccount:
```

Comment:
This is something can be done post GA. But it's will be confusing to SDK customers that they need to use SystemCreatedStorageAccount class for both system created and customer created storage accounts.

Reply:
Resolving this for now, will look into this post-ga.